adapters/repos/db/index.go

```diff
@@ -1724,71 +1721,102 @@ func (i *Index) IncomingDeleteObject(ctx context.Context, shardName string,
 	return shard.DeleteObject(ctx, id)
 }
 
-// func (i *Index) localShard(name string) ShardLike {
-// 	return i.shards.Load(name)
-// }
+// Intended to run on "receiver" nodes, where local shard
+// is expected to exist and be active
+// Method first tries to get shard from Index::shards map,
+// or inits shard and adds it to the map if shard was not found
+func (i *Index) getOrInitLocalShard(ctx context.Context, shardName string) (ShardLike, error) {
+	className := i.Config.ClassName.String()
+	class := i.getSchema.ReadOnlyClass(className)
+	return i.initLocalShard(ctx, class, shardName)
+}
+
+func (i *Index) initLocalShard(ctx context.Context, class *models.Class, shardName string) (ShardLike, error) {
+	return i.optInitLocalShard(ctx, class, shardName, !i.Config.DisableLazyLoadShards)
+}
 
-func (i *Index) getLocalShardNoShutdown(shardName string) (ShardLike, func(), error) {
-	i.shardInUseLocks.RLock(shardName)
-	defer i.shardInUseLocks.RUnlock(shardName)
+func (i *Index) loadLocalShard(ctx context.Context, shardName string) error {
+	className := i.Config.ClassName.String()
+	class := i.getSchema.ReadOnlyClass(className)
+	_, err := i.optInitLocalShard(ctx, class, shardName, true)
+	return err
+}
 
-	shard := i.shards.Load(shardName)
-	if shard == nil {
-		return nil, func() {}, nil
+func (i *Index) optInitLocalShard(ctx context.Context, class *models.Class, shardName string, disableLazyLoad bool) (ShardLike, error) {
```

Comment:
nit: What is the purpose of the `opt` prefix here ? for me this is getting the shard or loading it, do you mean optional ? In which case would it be optional ?